src/server/controlchan/commands/pass.rs

```diff
@@ -73,10 +76,33 @@ where
                     source_ip: session.source.ip(),
                     certificate_chain: session.cert_chain.clone(),
                 };
+                let failed_logins = session.failed_logins.clone();
+                let source_ip = session.source.ip();
                 tokio::spawn(async move {
                     let msg = match auther.authenticate(&username, &creds).await {
                         Ok(user) => {
-                            if user.account_enabled() {
+                            let is_locked = match failed_logins {
+                                Some(failed_logins) => {
+                                    let result = failed_logins.success(source_ip, username.clone()).await;
+                                    if let Err(err) = result {
+                                        slog::warn!(
+                                            logger,
+                                            "User authenticated but currently locked out due to previous failed login attempts according to the policy! (Username={}. Note: the account automatically unlocks after the configured period if no further failed login attempts occur. Error: {:?})",
+                                            username,
+                                            err
+                                        );
+                                        true
+                                    } else {
+                                        false
+                                    }
+                                }
+                                None => false,
+                            };
+
+                            if is_locked {
+                                sleep(Duration::from_millis(1500)).await;
```

Comment:
Should this penalty perhaps also be configurable?

Reply:
I agree but this is also hardcoded in the json file auth currently. It should not be in there either, but that's how it now works.
I think we need to refactor this whole thing.